osu.Server.Queues.ScoreStatisticsProcessor/Processors/ManiaKeyModeUserStatsProcessor.cs

```diff
@@ -94,24 +62,75 @@ public void ApplyToUserStats(SoloScore score, UserStats fullRulesetStats, MySqlC
                             userId = score.user_id,
                             rulesetId = score.ruleset_id,
                             keyCount = keyCount,
-                        }, transaction) ?? 1;
-
-                    conn.Execute(
-                        $"REPLACE INTO `{keyCountTableName}` "
-                        + $"(`user_id`, `country_acronym`, `playcount`, `x_rank_count`, `xh_rank_count`, `s_rank_count`, `sh_rank_count`, `a_rank_count`, `rank_score`, `rank_score_index`, `accuracy_new`) "
-                        + $"SELECT @user_id, `country_acronym`, @playcount, @x_rank_count, @xh_rank_count, @s_rank_count, @sh_rank_count, @a_rank_count, @rank_score, @rank_score_index, @accuracy_new "
-                        + $"FROM `phpbb_users` WHERE `user_id` = @user_id",
-                        keymodeStats, transaction);
-                }
+                        }, transaction) ?? 0,
+                };
+
+                conn.Execute(
+                    $"REPLACE INTO `{keyCountTableName}` "
+                    + $"(`user_id`, `country_acronym`, `playcount`, `x_rank_count`, `xh_rank_count`, `s_rank_count`, `sh_rank_count`, `a_rank_count`, `rank_score`, `rank_score_index`, `accuracy_new`, `last_played`) "
+                    + $"SELECT @user_id, `country_acronym`, @playcount, @x_rank_count, @xh_rank_count, @s_rank_count, @sh_rank_count, @a_rank_count, @rank_score, @rank_score_index, @accuracy_new, @last_played "
+                    + $"FROM `phpbb_users` WHERE `user_id` = @user_id",
+                    keyModeStats, transaction);
+            }
+            else
+            {
+                keyModeStats.playcount++;
+            }
+
+            if (score.preserve)
+            {
+                updateRankCounts(score, keyModeStats, conn, transaction);
+                UpdateUserStatsAsync(keyModeStats, keyCount, conn, transaction).Wait();
+
+                conn.Execute(
+                    $"UPDATE `{keyCountTableName}` "
+                    + $"SET `rank_score` = @rank_score, `playcount` = @playcount, `rank_score_index` = @rank_score_index, `accuracy_new` = @accuracy_new, "
+                    + $"`x_rank_count` = @x_rank_count, `xh_rank_count` = @xh_rank_count, `s_rank_count` = @s_rank_count, `sh_rank_count` = @sh_rank_count, `a_rank_count` = @a_rank_count "
+                    + $"WHERE `user_id` = @user_id", keyModeStats, transaction);
             }
             else
             {
-                conn.Execute($"UPDATE `{keyCountTableName}` SET `playcount` = `playcount` + 1 WHERE `user_id` = @user_id", keymodeStats, transaction);
+                conn.Execute($"UPDATE `{keyCountTableName}` SET `playcount` = @playcount, `last_played` = NOW() WHERE `user_id` = @user_id", keyModeStats, transaction);

```

Comment:
Why is `last_played` seemingly only updated in this branch? Shouldn't it also be getting set in the `score.preserve` branch?

Reply:
good catch, got lost in one of my refactors.

Reply:
Fixed via 83bea01